src/properties/text_property.rs

```diff
@@ -1,292 +1,1043 @@
-        validate!(
-            cursor,
-            component_type <= 2 || component_type == 8,
-            "Unexpected component {component_type}"
-        );
-
-        let expect_indicator = match component_type {
-            1 => 3,
-            8 => 0,
-            _ => 255,
-        };
-        let indicator = cursor.read_u8()?;
-        validate!(
-            cursor,
-            indicator == expect_indicator,
-            "Unexpected indicator {} for component {}, expected {}",
-            indicator,
-            component_type,
-            expect_indicator
-        );
-
-        if component_type == 0 {
-            // Empty text
-            let count = cursor.read_u32::<LittleEndian>()?;
-            validate!(cursor, count == 0, "Unexpected count {count}");
-
-            Ok(TextProperty::Empty([]))
-        } else if component_type == 1 {
-            // Rich text
-            let num_flags = cursor.read_u8()?;
-            validate!(cursor, num_flags == 8, "Unexpected num_flags {num_flags}");
-            let flags = cursor.read_u64::<LittleEndian>()?;
-            let expect_flags = if options.large_world_coordinates {
-                RTF_UNKNOWN
-            } else {
-                0
-            };
-            validate!(cursor, flags == expect_flags, "Unexpected flags {flags:X}");
-
-            if flags == RTF_UNKNOWN {
-                let b = cursor.read_u8()?;
-                assert_eq!(b, 0);
-            }
-
-            let id = cursor.read_string()?;
-            let pattern = cursor.read_string()?;
-            let arg_count = cursor.read_u32::<LittleEndian>()?;
-
-            let mut text_format = vec![];
-            for _ in 0..arg_count {
-                let format_key = cursor.read_string()?;
-                let separator = cursor.read_u8()?;
-                validate!(cursor, separator == 4, "Unexpected separator {separator}");
-                let content_type = cursor.read_u32::<LittleEndian>()?;
-                let indicator = cursor.read_u8()?;
-                validate!(cursor, indicator == 255, "Unexpected indicator {indicator}");
-                let count = cursor.read_u32::<LittleEndian>()?;
-
-                let mut values = vec![];
+        let start = cursor.stream_position()?;
+        let history_type =
+            TextHistoryType::try_from(cursor.read_i8()?).map_err(|e| DeserializeError::from(e))?;
+
+        Ok(match history_type {
+            TextHistoryType::None => {
+                let culture_invariant_string = if options
+                    .get_custom_version::<FEditorObjectVersion>()
+                    .version
+                    >= FEditorObjectVersion::CultureInvariantTextSerializationKeyStability as u32
+                {
+                    let has_culture_invariant_string = cursor.read_u32::<LittleEndian>()? != 0;
+                    if has_culture_invariant_string {
+                        Some(cursor.read_fstring()?)
```

Comment:
Is it possible to get a `Some(None)` here?

Reply:
It is, that's when it's present but the value of the FString serialized from the engine null